Branch-SDK/Branch-SDK/BranchActivityItemProvider.m

```diff
@@ -31,15 +31,27 @@ - (id)initWithParams:(NSDictionary *)params
         self.branchURL = url;
     }
     
+    dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
You're dispatching *to* the main thread (read: locking the main thread) here. Any reason you're trying to push this into a block instead of just evaluating directly?